app/controller/button/DigitizeButtonController.js

```diff
@@ -0,0 +1,479 @@
+        }
+        me.doAjaxRequest(jsonParams, searchParams);
+    },
+
+    /**
+     * Issues an Ext.Ajax.request against the configured endpoint with
+     *   the given params. Returned features will be added to the result
+     *   layer, and the drawn features will be removed. Additionally,
+     *   an event is fired with the response features to allow other
+     *   components to make use of them.
+     * @param {object} jsonParams Object containing the params to send
+     * @param {string} searchParams The serarchParams which will be
+     *   appended to the request url
+     */
+    doAjaxRequest: function (jsonParams, searchParams) {
+        var me = this;
+        var mapComponent = BasiGX.util.Map.getMapComponent();
+        var view = me.getView();
+        var url = view.getApiUrl();
+        var format = new ol.format.GeoJSON();
+        var responseData;
+        if (!url) {
+            return;
+        }
+        if (searchParams) {
+            url += '?' + searchParams;
+        }
+        mapComponent.setLoading(true);
+        Ext.Ajax.request({
+            url: url,
+            method: 'POST',
+            jsonData: jsonParams,
+            callback: function() {
+                mapComponent.setLoading(false);
+            },
+            success: function(response){
+                if(!Ext.isEmpty(response.responseText)) {
+                    var json = Ext.decode(response.responseText);
+                    if (json.success) {
+                        responseData = json.data;
+                        if (json.data && json.data.features) {
+                            var olFeatsForActiveGroup = [];
+                            Ext.each(json.data.features, function(feat) {
+                                // api will respond with non unique ids, deleting it here...
+                                delete feat.id;
+                                feat.properties.group = me.activeGroupIdx;
+                                olFeatsForActiveGroup.push(format.readFeature(feat));
+                            });
+                            responseData = olFeatsForActiveGroup;
+                            var allFeatures = me.resultLayer.getSource().getFeatures();
+                            Ext.each(allFeatures, function(f){
+                                if (f.get('group') === me.activeGroupIdx) {
+                                    me.resultLayer.getSource().removeFeature(f);
+                                }
+                            });
+                            me.resultLayer.getSource().addFeatures(olFeatsForActiveGroup);
+                            var drawSource = me.drawLayer.getSource();
+                            if (view.getClearDrawnFeature()) {
+                                drawSource.clear();
+                            } else if (drawSource.getFeatures().length > 1) {
+                                me.drawLayer.getSource().removeFeature(drawSource.getFeatures()[0]);
```

Comment:
Just for clarification: Is here always just 1 feature in the source?

Reply:
No, it may happen that a single old drawn feature is there. We will remove the old (first) feature everytime and just keep the new one.